File: src/easy/valid_mountain_array.zig
const std = @import("std");

pub fn validMountainArray(comptime T: type, arr: []const T) bool {
    var count: usize = 1;

    while (count < arr.len and arr[count] > arr[count - 1]) count += 1;

    if (count == 1 or count == arr.len) return false;
    while (count < arr.len and arr[count] < arr[count - 1]) count += 1;
    return count == arr.len;
}

File: easy/valid_mountain_array.go
package easy

func ValidMountainArray(arr []int) bool {
	n := len(arr)
	if n < 3 {
		return false
	}
	i := 0
	for i+1 < n && arr[i] < arr[i+1] {
		i++
	}
	if i == 0 || i == n-1 {
		return false
	}
	for i+1 < n && arr[i] > arr[i+1] {
		i++
	}
	return i == n-1
}
